### DISCUSSION TITLE
Release 1.25.1 for MacOS no longer a single executable

### DISCUSSION THREAD

**Discussion by suroh1994** [layday/instawow#80] [Q&A]
As the title suggests, the latest release zip for MacOS does not contain a single executable, but instead the executable and a lib folder with a bunch of .so files. Is this the new intended structure?

Once I unpacked the archive and attempted to launch instawow, the app was blocked because it is not from an identified developer. After going into settings > Security & Privacy and allowing the app to run, I can launch it, but it will be blocked again, telling me to allow some .so file to be launched. This process repeats a bunch of times and after like 5 times I just gave up and reverted to 1.24.0.

Is there a good solution to this or should I move to installing via pip instead?

**Comment by layday** [comment]
It's the structure imposed on us by [PyOxidizer](https://github.com/indygreg/PyOxidizer), the program used to bundle the CLI.  Older versions did produce a single executable but without extension modules.  The new version is able to link extension modules but it can't load them from memory.  To get the CLI to run, you can do the following, assuming you are in a folder with just `instawow` and `instawow-lib`:

```
$ chmod +x instawow
$ find . -type f | xargs xattr -d com.apple.quarantine
# verify that it works
$ ./instawow --version
```

**Reply by suroh1994** [reply]
Alright. Your steps worked, but I would argue it would be nice to have them somewhere in the readme, as others might also run into this issue. I would not have known how to fix this, so thanks for your help!

**Reply by layday** [reply]
I agree that a note in the readme would be good. What would be even better is if users didn't have perform this incantation.
